Scheduled Updates: Include name param and update schema

obenland opened the pull request:
## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Accept name parameter to store alongside plugins.
* Update response schema to include information needed to display schedules in Calypso.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

TBD

obenland committed:
Scheduled Updates: Include name param and update schema

vishnugopal commented on projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules.php:
```diff
@@ -184,7 +198,10 @@ public function create_item( $request ) {
+		);
 		update_option( 'jetpack_update_schedules', $schedules );
 
 		return rest_ensure_response( $this->generate_schedule_id( $plugins ) );
```
It'd be nice if we can store the `id` in `jetpack_update_schedules` too and return it in `get_item` and `get_items`. I found it a bit strange that we return an id for create, but not for get or list.

bogiii replied:
+1, without ID... I can not delete or update items.

ouikhuan commented:
Some thoughts about the `generate_schedule_id` function, we might want to make sure the plugins come in the same order every single time and sort it. For example, the result of the following will be different:
```
Input:
$plugins = array(
	'gutenberg/gutenberg.php',
	'custom-plugin/custom-plugin.php',
);
Output: 0e8084c8c932e72b700da80f81c153ff
```

```
Input:
$plugins = array(
	'custom-plugin/custom-plugin.php',
	'gutenberg/gutenberg.php',
);
Output: 6e5d5fed314ed0b81b54723ef1489c29
```
But they have the same plugins checked.

obenland commented:
@Automattic/caribou Could you see if https://github.com/Automattic/jetpack/pull/36071 addresses some of the issues you mentioned?

It sorts plugin arrays, gets rid of the `jetpack_update_schedules` option entirely, and includes the hash id in `get_items()` responses. It's quite a bit more elegant, I think.

obenland commented:
Closing in favor of https://github.com/Automattic/jetpack/pull/36071
